community/kernel/src/main/java/org/neo4j/kernel/impl/store/record/AbstractSchemaRule.java

```diff
@@ -40,10 +54,9 @@ public long getId()
         return this.id;
     }
 
-    @Override
-    public final int getLabel()
+    protected final int getLabelOrRelationshipType()
```

Comment:
This doesn't feel quite right to me. I think I'd prefer if the fields for this data was defined in the sub-types that actually know if they have a `label` field or a `relationshipType` field. The cost of this would be that the `serialize` method would have to be abstract in this class, but I think that is acceptable.